shared/teams/team/member/add-to-channels.tsx

```diff
@@ -253,20 +254,35 @@ const SelfChannelActions = ({
 
   const actionProps = {conversationIDKey: meta.conversationIDKey, teamID: meta.teamID}
   const onEditChannel = () => {} // TODO: show another modal with a back button to here
-  const onAdministrate = () =>
+  const onAdministrate = () => {
+    dispatch(RouteTreeGen.createClearModals())
     dispatch(nav.safeNavigateAppendPayload({path: [{props: actionProps, selected: 'teamChannel'}]}))
+  }
   const onDelete = () =>
-    // TODO: reload this list or at least hide this row
+    // TODO: consider not using the confirm modal
     dispatch(nav.safeNavigateAppendPayload({path: [{props: actionProps, selected: 'teamDeleteChannel'}]}))
 
   const joinRPC = Container.useRPC(RPCChatGen.localJoinConversationByIDLocalRpcPromise)
   const leaveRPC = Container.useRPC(RPCChatGen.localLeaveConversationLocalRpcPromise)
 
-  const waitingKey = `joinOrLeave:${meta.conversationIDKey}`
-  const waiting = Container.useAnyWaiting(waitingKey)
+  const [waiting, setWaiting] = React.useState(false)
   const convID = ChatTypes.keyToConversationID(meta.conversationIDKey)
-  const onLeave = () => leaveRPC([{convID}, waitingKey], triggerReload, () => {})
-  const onJoin = () => joinRPC([{convID}, waitingKey], triggerReload, () => {})
+  const onLeave = () => {
+    setWaiting(true)
+    leaveRPC(
+      [{convID}],
+      () => reloadChannels().then(() => setWaiting(false)),
+      () => {}
```

Comment:
`setWaiting(false)` on these. also we need `catch`s on the reloadChannels calls here or I think they'll throw.